src/resource-metadata/lambda.js

```diff
@@ -0,0 +1,214 @@
+const assert = require("assert");
+const { LambdaClient, ListFunctionsCommand, GetFunctionCommand, ListAliasesCommand, GetPolicyCommand, ListVersionsByFunctionCommand, ListEventSourceMappingsCommand } = require("@aws-sdk/client-lambda");
+const pLimit = require('p-limit');
+const pThrottle = require('p-throttle');
+const { schemaUrl, extractArchitecture, intAttr, stringAttr, traverse } = require("./common.js")
+
+const lambdaClient = new LambdaClient();
+
+assert(process.env.LAMBDA_API_RATE_LIMIT, "LAMBDA_API_RATE_LIMIT env var missing!")
+const rateLimit = parseInt(process.env.LAMBDA_API_RATE_LIMIT, 10)
+
+assert(process.env.LATEST_VERSIONS_PER_FUNCTION, "LATEST_VERSIONS_PER_FUNCTION env var missing!")
+const latestVersionsPerFunction = parseInt(process.env.LATEST_VERSIONS_PER_FUNCTION, 10)
+
+assert(process.env.RESOURCE_TTL_MINUTES, "RESOURCE_TTL_MINUTES env var missing!")
+const resourceTtlMinutes = parseInt(process.env.RESOURCE_TTL_MINUTES, 10)
+
```

Comment:
It can be in a different config file.
```suggestion
function getConfig() {
    assert(process.env.LAMBDA_API_RATE_LIMIT, "LAMBDA_API_RATE_LIMIT env var missing!")
    const rateLimit = parseInt(process.env.LAMBDA_API_RATE_LIMIT, 10)

    assert(process.env.LATEST_VERSIONS_PER_FUNCTION, "LATEST_VERSIONS_PER_FUNCTION env var missing!")
    const latestVersionsPerFunction = parseInt(process.env.LATEST_VERSIONS_PER_FUNCTION, 10)

    assert(process.env.RESOURCE_TTL_MINUTES, "RESOURCE_TTL_MINUTES env var missing!")
    const resourceTtlMinutes = parseInt(process.env.RESOURCE_TTL_MINUTES, 10)
    return {rateLimit, latestVersionsPerFunction, resourceTtlMinutes};
}

const {rateLimit, latestVersionsPerFunction, resourceTtlMinutes} = getConfig();

```